packages/angular_devkit/schematics/src/engine/engine.ts

```diff
@@ -78,9 +79,10 @@ export class SchematicEngine<CollectionT extends object, SchematicT extends obje
     = new Map<string, Map<string, SchematicImpl<CollectionT, SchematicT>>>();
   private _taskSchedulers = new Array<TaskScheduler>();
 
-  constructor(private _host: EngineHost<CollectionT, SchematicT>) {
+  constructor(private _host: EngineHost<CollectionT, SchematicT>, protected _workflow?: Workflow) {
   }
 
+  get workflow() { return this._workflow || null; }
```

Comment:
Exposing this downward into the implementation interweaves the higher level abstraction of the `Workflow` with the rest of the system.  I think ideally a schematic or task shouldn't have to rely on the Workflow or that a Workflow even exists.

Some ideas on how this could be implemented in the comments for the node workflow.

Reply:
Post 6.0. I put this property as `@internal` so at least it won't be in the `.d.ts` files.